src/components/DownloadForm/Controls/index.js

```diff
@@ -56,14 +57,17 @@ export class Controls extends PureComponent /*:: <Props> */ {
     noData: T.bool.isRequired,
   };
 
-  _handleGenerateClick = blockEvent(() =>
+  _handleGenerateClick = blockEvent(() => {
+    // Request browser notification
+    askNotificationPermission(() => {});
```

Comment:
Minor comment, but I think It's kind of weird to have to pass an empty function. An alternative is to use `noop` from lodash as the default so if you don't provide `setState` it calls `noop`.

I did something like it here: https://github.com/ProteinsWebTeam/interpro7-client/blob/b3b6cdead118b886e163764bdfc262d3d54253df/src/components/SimpleCommonComponents/Modal/index.js

Reply:
Thanks for the comment Gustavo. I will change it.